### PR TITLE
Add a function to add an ssh key to a repo

### PR DESCRIPTION
**Pull request by ttetzlaff**

### CODE REVIEW

File: vcsclient/bitbucketcloud.go [hunk 1]
```diff
@@ -72,6 +73,61 @@ func (client *BitbucketCloudClient) ListBranches(ctx context.Context, owner, rep
 	return results, nil
 }
 
+// AddSshKeyToRepository on Bitbucket cloud the deploy-key is always read only
+func (client *BitbucketCloudClient) AddSshKeyToRepository(ctx context.Context, owner, repository, keyName, publicKey string, _ Permission) error {
+	// https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/deploy-keys#post
```
Comment (yahavi) [inline vcsclient/bitbucketcloud.go:78]: We don't like URLs in the code
```suggestion
```

File: vcsclient/bitbucketcloud.go [hunk 2]
```diff
@@ -72,6 +73,61 @@ func (client *BitbucketCloudClient) ListBranches(ctx context.Context, owner, rep
...
+	req.Header.Set("Content-Type", "application/json")
+	req.SetBasicAuth(client.vcsInfo.Username, client.vcsInfo.Token)
+
+	bitbucketClient := client.buildBitbucketCloudClient(ctx)
+	response, err := bitbucketClient.HttpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = vcsutils.DiscardResponseBody(response)
```
Comment (yahavi) [inline vcsclient/bitbucketcloud.go:116]: What is the reason for consuming the response body before closing it?

File: vcsclient/bitbucketcloud.go [hunk 3]
```diff
@@ -72,6 +73,61 @@ func (client *BitbucketCloudClient) ListBranches(ctx context.Context, owner, rep
 	return results, nil
 }
 
+// AddSshKeyToRepository on Bitbucket cloud the deploy-key is always read only
```
Comment (yahavi) [inline vcsclient/bitbucketcloud.go:76]:
```suggestion
// AddSshKeyToRepository on Bitbucket cloud. The deploy-key is always read-only.
```

**Review by yahavi — CHANGES_REQUESTED** [review]
Please also add a description in the README for AddSshKeyToRepository

File: vcsclient/bitbucketcloud.go (hunk 2 above)
Comment (ttetzlaff) [reply vcsclient/bitbucketcloud.go:116]: We ran into a problem once where the connection wasn't closed correctly when the body wasn't consumed.

File: vcsclient/bitbucketcloud.go (hunk 3 above)
Comment (ttetzlaff) [reply vcsclient/bitbucketcloud.go]: How about 
` // AddSshKeyToRepository on Bitbucket cloud, the deploy-key is always read-only.`

File: vcsclient/bitbucketcloud.go (hunk 2 above)
Comment (cyrilc-pro) [reply vcsclient/bitbucketcloud.go:115]: From http.Client.Do() documentation:

If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close. If the Body is not both read to EOF and closed, the Client's underlying RoundTripper (typically Transport) may not be able to re-use a persistent TCP connection to the server for a subsequent "keep-alive" request.

File: vcsclient/bitbucketcloud.go (hunk 3 above)
Comment (yahavi) [reply vcsclient/bitbucketcloud.go]: Looks good

**Pull request merged**